src/main/java/frc/robot/subsystems/IndexerSubsystem.java

```diff
@@ -1,126 +1,133 @@
 import static frc.robot.Constants.IndexerConstants.DEVICE_ID_INDEXER_RIGHT_MOTOR;
-import static frc.robot.Constants.IndexerConstants.THRESHOLD_INTAKE;
-import static frc.robot.Constants.IndexerConstants.THRESHOLD_SHOOTER;
-
-import java.util.Map;
 
 import com.revrobotics.CANSparkBase.ControlType;
 import com.revrobotics.CANSparkBase.IdleMode;
 import com.revrobotics.CANSparkMax;
+import com.revrobotics.ColorMatch;
+import com.revrobotics.ColorSensorV3;
 import com.revrobotics.SparkPIDController;
 
-import edu.wpi.first.wpilibj.Notifier;
-import edu.wpi.first.wpilibj.shuffleboard.BuiltInLayouts;
-import edu.wpi.first.wpilibj.shuffleboard.BuiltInWidgets;
+import edu.wpi.first.wpilibj.I2C.Port;
 import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardLayout;
 import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
 import edu.wpi.first.wpilibj2.command.Command;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
 import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine.Direction;
 import frc.robot.Constants.IndexerConstants;
-import frc.robot.commands.ColorSensorReader;
 import frc.robot.subsystems.sysid.SysIdRoutineSignalLogger;
 
 public class IndexerSubsystem extends SubsystemBase {
-  public enum IndexerState {
-    noRing,
-    intakingRing,
-    HasRing,
-    processingRing
-  }
+  private final CANSparkMax leftIndexerMotor = new CANSparkMax(DEVICE_ID_INDEXER_LEFT_MOTOR, kBrushless);
+  private final CANSparkMax rightIndexerMotor = new CANSparkMax(DEVICE_ID_INDEXER_RIGHT_MOTOR, kBrushless);
+  private static SparkPIDController leftPidController;
+  private static SparkPIDController rightPidController;
 
-  public IndexerState indexerState;
+  private final ColorSensorV3 indexerColorSensor = new ColorSensorV3(Port.kOnboard);
+  private final ColorMatch indexerColorMatch = new ColorMatch();
 
-  public boolean capturedRing() {
-    return (this.indexerState != IndexerState.noRing);
-  }
 
-  public IndexerState getIndexerState() {
-    return indexerState;
-  }
+  public IndexerSubsystem() {
+    leftPidController = indexerMotorConfig(leftIndexerMotor, true);
+    rightPidController = indexerMotorConfig(rightIndexerMotor, false);
 
-  private boolean isIntakeSensorTripped() {
-    return colorSensorReader.getIntakeValues().proximity > THRESHOLD_INTAKE;
+    indexerColorMatch.addColorMatch(COLOR_NOTE);
+    indexerColorMatch.addColorMatch(COLOR_NONE);
   }
-
-  public boolean isShooterSensorTripped() {
-    return colorSensorReader.getShooterValues().proximity > THRESHOLD_SHOOTER;
+  
+  public boolean shouldContinue() {
+    return indexerColorMatch.matchClosestColor(indexerColorSensor.getColor()).color == COLOR_NOTE;
   }
 
-  public void stop() {
-    pidController.setReference(0, ControlType.kVelocity);
+  // Check actual motor values for dash board debugging
+  public boolean isActive() {
+    return leftIndexerMotor.get() > 0 || rightIndexerMotor.get() > 0;
```

Comment:
I think `get()` only returns a non-zero value if you're running with duty cycle.
I'm pretty sure `getAppliedOutput()` or `getOutputCurrent()` will work with closed loop control, but you'll have to test it.